src/main/scala/edu/ie3/simona/model/thermal/ThermalGrid.scala

```diff
@@ -399,40 +403,45 @@ final case class ThermalGrid(
 
   /** Convert the given state of the thermal grid into result models of it's
     * constituent models
+    *
+    * @param currentTick
+    *   Actual simulation tick
     * @param state
     *   State to be converted
     * @param startDateTime
     *   Start date time of the simulation
     * @return
     *   A [[Seq]] of results of the constituent thermal model
     */
-  def results(
-      state: ThermalGridState
-  )(implicit startDateTime: ZonedDateTime): Seq[ResultEntity] = {
-    val houseResults = house
+  def results(currentTick: Long, state: ThermalGridState)(implicit
+      startDateTime: ZonedDateTime
+  ): Seq[ResultEntity] = {
+
+    val maybeHouseResult = house
       .zip(state.houseState)
+      .filter { case (_, state) => state.tick == currentTick }
```

Comment:
Looks to me that this rather should be lastTick. Since we yet haven't any ThermalGridIT this comes not directly clear.

' .filter { case (_, state) => state.tick == lastTick}'